pkg/client/client.go

```diff
@@ -1122,6 +1134,18 @@ func monitorKeepTerminatedEvent(
 						logger.Warningf("Please broadcast Bitcoin transaction %s", recoveryTransactionHex)
 						logger.Warningf("Please broadcast Bitcoin transaction %s", recoveryTransactionHex)
 
+						if bitcoinConfig.BeneficiaryAddress != beneficiaryAddress {
+							err = derivationIndexStorage.Save(bitcoinConfig.BeneficiaryAddress, addressIndex)
+							if err != nil {
+								logger.Errorf(
+									"failed to persist the latest address derivation index for keep [%s]: [%v]",
+									keep.ID(),
+									err,
+								)
+								return err
```

Comment:
As we return the error this means we will retry the execution of the whole function, right? Maybe we should just log an error (without return) and let the process complete?